presto-hive/src/main/java/io/prestosql/plugin/hive/HivePageSource.java

```diff
@@ -173,7 +174,7 @@ public HivePageSource(
                 coercers.add(Optional.empty());
             }
 
-            if (columnMapping.getKind() == EMPTY) {
+            if (columnMapping.getKind() == EMPTY || isRowIdColumnHandle(column)) {
```

Comment:
It isn't redundant, because unlike other "prefilled columns", this column requires row-level data from the TableScan, so it it filled differently than prefilled columns.